src/poliastro/twobody/orbit.py

```diff
@@ -389,30 +389,37 @@ def from_body_ephem(cls, body, epoch=None):
                 TimeScaleWarning,
             )
 
-        r, v = get_body_barycentric_posvel(body.name, epoch)
-
-        if body == Moon:
-            # TODO: The attractor is in fact the Earth-Moon Barycenter
-            icrs_cart = r.with_differentials(v.represent_as(CartesianDifferential))
-            gcrs_cart = (
-                ICRS(icrs_cart)
-                .transform_to(GCRS(obstime=epoch))
-                .represent_as(CartesianRepresentation)
-            )
-            ss = cls.from_vectors(
-                Earth,
-                gcrs_cart.xyz.to(u.km),
-                gcrs_cart.differentials["s"].d_xyz.to(u.km / u.day),
-                epoch,
-            )
+        try:
+            r, v = get_body_barycentric_posvel(body.name, epoch)
+            if body == Moon:
```

Comment:
Instead of wrapping everything in a `try ... except`, I would wrap only the `r, v = ...` line, and catch the `KeyError` there.